### PR TITLE
4.x - Added migration notes for DateTimeTimezoneType

### PR DESCRIPTION
**Pull request by othercorey**
Migration notes for https://github.com/cakephp/cakephp/pull/13943

### CODE REVIEW

File: en/appendices/4-0-migration-guide.rst [hunk 1]
```diff
@@ -173,6 +173,17 @@ Database
     #. ``DATETIME2`` => ``datetimefractional``
     #. ``DATETIME2(1-7) => ``datetimefractional``
 
+* PostgreSQL schema now map columns that support time zones to the new abstract
+  time zone types. Specifying (0) precision does not change the type mapping like
+  it does with regular frectional types above.
```
Comment (markstory) [inline en/appendices/4-0-migration-guide.rst]:
```suggestion
  it does with regular fractional types above.
```

File: en/appendices/4-0-migration-guide.rst [hunk 2]
```diff
@@ -389,9 +400,10 @@ Database
 * The ``char`` abstract type was added. This type handles fixed length string
   columns.
 * The ``datetimefractional`` and ``timestampfractional`` abstract types were added.
-  These types handle timestamps with fractional seconds.
+  These types handle column data types with fractional seconds.
 * SqlServer schemas now support default values with functions in them like SYSDATETIME().
-
+* The ``datetimetimezone`` and ``timestmaptimezone`` abstract types were added.
```
Comment (markstory) [inline en/appendices/4-0-migration-guide.rst]:
```suggestion
* The ``datetimetimezone`` and ``timestamptimezone`` abstract types were added.
```

File: en/appendices/4-0-migration-guide.rst [hunk 3]
```diff
@@ -452,6 +464,11 @@ ORM
   the ``TypeFactory`` as the default ``datetime`` type or re-mapping individual
   columns. See the Database migration notes for how this type is automatically
   mapped to database types.
+* A new type class ``DateTimeTimezoneType`` has been added for datetime types
+  with time zone support. You can opt into using this type by adding it to
+  the ``TypeFactory`` as the default ``datetime`` type or re-mapping individual
+  columns. See the Database migration notes for how this type is automatically
+  mapped to database types.
```
Comment (markstory) [inline en/appendices/4-0-migration-guide.rst]: I don't know if these types will be supported by migrations, we might want to figure that out as well.

Comment (othercorey) [reply en/appendices/4-0-migration-guide.rst]: Confirmed timezone type is in phinx postgres config.

**Comment by emaillenin** [comment]
When will the type `datetimetimezone` be supported for MySQL?

> Phinx\Db\Adapter\UnsupportedColumnTypeException: Column type "datetimetimezone" is not supported by MySQL

**Comment by garas** [comment]
MySQL can only store timestamps without timezone information. Only PostgreSQL and SQL Server has such feature.

**Comment by othercorey** [comment]
@emaillenin That phinx error is unrelated to the new types in CakePHP 4. Like garas said, the type simply doesn't exist in mysql.

**Pull request merged**